softwareComponents/shapeReconfig/include/shapeReconfig/hashing.hpp

```diff
@@ -0,0 +1,50 @@
+#include <configuration/rofiworld.hpp>
+
+#include <shapeReconfig/geometry.hpp>
+
+namespace rofi::shapereconfig {
+
+size_t combineHash( size_t h1, size_t h2 );
+
+struct RofiWorldHash
+{
+    size_t operator()( const rofi::configuration::RofiWorld& rw ) const;
+};
+
+template < typename _Tp, size_t _Nm >
+struct HashArray
+{
+    size_t operator()( const std::array< _Tp, _Nm >& arr ) const
+    {
+        size_t h = 0;
+        for ( const _Tp& x : arr )
```

Comment:
Use can use `std::accumulate`:
`std::accumulate(arr.begin(), arr.end(), size_t{}, combineHash)` (note that the init cannot be 0 - or it will infer int as the type).

Reply:
There is no conversion to `size_t` - the input function is essentially of type `(size_t, T) -> size_t` where T can be arbitrary - and is the type over which the iterators iterate. Specifically it doesn't have to be convertible to `size_t`.

Note: it's not this simple because it allows conversions, but there is no conversion between the output type and the input iterated type.

Reply:
Sorry, I read it wrong - but you could write it as something like:
```
auto tranArr = std::ranges::transform( arr, std::hash< T >{} );
return std::accumulate( tranArr.begin(), tranArr.end(), size_t{}, combineHash);
```
(Not sure if the functor works - if not, you can use lambda.)

Reply:
(This is the same as with hashing the cloud.)